2_process/src/process_precip.R

```diff
@@ -0,0 +1,68 @@
+
+process_precip_rasters <- function(ind_file, precip_spatial_ind,
+                                   precip_values_ind,
+                                   view_polygon,
+                                   view_config) {
+
+  precip_spatial <- readRDS(sc_retrieve(precip_spatial_ind)) %>%
+    dplyr::select(-x, -y)
+
+  precip <- readRDS(sc_retrieve(precip_values_ind))
+
+
+  time <- dplyr::select(precip, time) %>%
+    distinct() %>%
+    arrange() %>%
+    mutate(time_id = 1:n())
+
+  precip <- left_join(precip, time, by = "time") %>%
+    dplyr::select(-time) %>%
+    group_by(id) %>%
+    arrange(time_id, .by_group = TRUE) %>%
+    mutate(precip = cumsum(precip)) %>%
+    ungroup()
+
+  ncol <- view_config$width
+
+  bbox <- sf::st_bbox(view_polygon)
+
+  y_range <- as.numeric(bbox$ymax - bbox$ymin)
+  x_range <- as.numeric(bbox$xmax - bbox$xmin)
+
+  aspect_ratio <- x_range / y_range
```

Comment:
aspect_ratio is present in viz_config.yml; is there any reason it'd be different from `view_config$aspect` by the time we get here? though as noted above, i'm hoping we can switch to `view_config$width` and `view_config$height` instead of keeping aspect in the config, in which case computing it from width and height would be fine. But still, can we skip the y_range and x_range lines, or are those needed for some reason?

Reply:
Yeah. This is messy from how I had it before. I'll get height / width in to view_config and clean house a bit.

Reply:
:+1: